### PR TITLE
Add 'disable campaign parameters tracking' setting

### PR DESCRIPTION
**Pull request by michalkleiner**
### Description:

Companion PR to https://github.com/matomo-org/matomo/pull/22013 enabling the same setting for MTM.

The new variable is available and when new version of the container is published after the checkbox is checked, it is correctly returned in the list of configurations for the container and campaign params are correctly stripped.

Removing the check, publishing a new version and tracking a link with campaign params after that works as expected again.

screenshot - disable campaign tracking MTM variable
screenshot - disable campaign tracking MTM

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by caddoo — APPROVED** [review]

File: Template/Tag/MatomoTag.web.js [hunk 1]
```diff
@@ -126,6 +126,10 @@
                         tracker.disableBrowserFeatureDetection();
                     }
 
+                    if (matomoConfig.disableCampaignParameters) {
```
Comment (snake14) [inline Template/Tag/MatomoTag.web.js]: I believe that this should match the name of the config created in `Template/Variable/MatomoConfigurationVariable.php`
```suggestion
                    if (matomoConfig.requireConsentForCampaignTracking) {
```

**Review by snake14 — CHANGES_REQUESTED** [review]
Aside from the one change request, this looks like it should work. I did functional testing with the change that I suggested and it appeared to work as expected.

File: Template/Tag/MatomoTag.web.js (hunk 1 above)
Comment (michalkleiner) [reply Template/Tag/MatomoTag.web.js]: Ok, thanks @snake14!

I wasn't sure as some of the other calls around this were referencing the methods and some were referencing the configs and I couldn't really find a pattern to follow. Will change this and see what the tests do. QA will test the overall change as well.

**Review by snake14 — APPROVED** [review]
Looks good to me :+1:

**Review by AltamashShaikh — APPROVED** [review]

**Comment by MatomoForumNotifications** [comment]
This pull request has been mentioned on **Matomo forums**. There might be relevant details there:

https://forum.matomo.org/t/new-tracking-functions-disablecampaignparameters-enablecampaignparameters/58176/1

**Pull request merged**